src/octoprint/util/__init__.py

```diff
@@ -502,6 +504,10 @@ def filter_non_ascii(line):
 
 def to_str(s_or_u, encoding="utf-8", errors="strict"):
 	"""Make sure ``s_or_u`` is a str."""
+	if PY3:
```

Comment:
Isn't this backwards? `to_str` should basically be `to_bytes` under Py3, so we'd need to check whether stuff is a `unicode` in 2 or `str` in 3 and if so decode, otherwise just return, or not?